src/NWCClient.ts

```diff
@@ -459,19 +459,31 @@ export class NWCClient {
     notifications: Nip47NotificationType[];
   }> {
     await this._checkConnected();
-
-    const events = await this.relay.list(
-      [
+    const events = await new Promise<Event[]>((resolve, reject) => {
+      const events: Event[] = [];
+      // TODO: test with different timeouts to see if this works
+      const sub = this.relay.subscribe(
+        [
+          {
+            kinds: [13194],
+            limit: 1,
+            authors: [this.walletPubkey],
+          },
+        ],
         {
-          kinds: [13194],
-          limit: 1,
-          authors: [this.walletPubkey],
+          eoseTimeout: 10000,
         },
-      ],
-      {
-        eoseSubTimeout: 10000,
-      },
-    );
+      );
+
+      sub.onevent = (event) => {
+        events.push(event);
+      };
+
+      sub.oneose = () => {
+        sub.close();
+        resolve(events);
+      };
+    });
```

Comment:
the behavior here is the same? we subscribe and on oneose the promise will be resolved? 
what's after the timeout?

Reply:
If the timeout is exceeded, the oneose function is called, which closes the subscription and resolves with events (`[]`)

Timeout function is added on firing in new relay: https://github.com/nbd-wtf/nostr-tools/blob/9c50b2c65515c6282872cb2cd8b4a58f7981c269/abstract-relay.ts#L349-L361

Old list method: https://github.com/nbd-wtf/nostr-tools/blob/318e3f8c8867cb183684622cdedf6a9ee49cee78/relay.ts#L305-L321

Reply:
> when do we actually need those eoseTimeout on the subscriptions?

I'm assuming when there are a lot of stored events that we keep fetching till the last event is reached, but in our case it's useless as we only have one persistent 13194 info event

Reply:
Right now it takes 10 seconds for the method to return, so I think if we get an event we should unsubscribe

Reply:
I did not run the code currently, but does it take 10 seconds? 
Shouldn't the relay send a eose and then the that function should fire?

Reply:
I checked the previous code and surprisingly the `eoseSubTimeout` wasn't even being used

> Right now it takes 10 seconds for the method to return

It returned almost instantaneously for me, it returns as soon as eose is reached no?

Reply:
True, that's right.

It's only 10 seconds in React Native in the latest version of Expo, (because of another issue which is also causing publishing to never return). I found a way to fix it in `nostr-tools` and will add a PR there.